File: frontend-mealsonwheels/src/types/admin/AdminStatisticDTO.ts
// src/types/management/AdminStatisticDTO.ts

export interface AdminStatisticDTO {
  totalMembers: number;
  totalPartners: number;
  totalVolunteers: number;
  totalRiders: number;
  totalDonors: number;
  mealsServed: number;
  ordersDelivered: number;
  totalDonationsReceived: string; // From BigDecimal
}

File: backend-mealsonwheels/src/main/java/com/merrymeal/mealsonwheels/dto/AdminStatisticDTO.java
package com.merrymeal.mealsonwheels.dto;

import lombok.*;

import java.math.BigDecimal;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AdminStatisticDTO {
    private long totalMembers;
    private long totalPartners;
    private long totalVolunteers;
    private long totalRiders;
    private long totalDonors;
    private long mealsServed;
    private long ordersDelivered;
    private BigDecimal totalDonationsReceived;
}
